doc/modules/model_evaluation.rst

```diff
@@ -1550,6 +1551,88 @@ Here is a small example of usage of this function:::
+
+The :func:`calibration_loss` function computes the expected and maximum
+calibration losses as defined in [1] for binary classes.
+
+Given a set of bins over predicted probabilities, the calibration loss
+measures the overall discrepancy per bin between
+(1) the average predicted probabilities assigned to the positive class, and
+(2) the frequencies of the positive class in the actual outcome in the same
+bin.
+
+The lower the calibration loss is for a set of predictions, the
+better the predictions are calibrated.
+
+The aggregation method ``norm`` can be either:
+
+- ``'l1'``: this computes :math:`\sum_k P_k \delta_k`.
+  This is the expected calibration error (ECE) defined in [1].
+- ``'l2'``: this computes :math:`\sqrt{\sum_k P_k \delta_k^2}`
+  the square root of the squared calibration error with
+  the "plugin" estimator if ``reduce_bias`` is set to ``False``, or
+  or with the "debiased" estimator (adding a non-positive term in the sum)
+  if ``reduce_bias`` is set to ``True`` [3].
+- ``'max'`` this computes :math:`\max_k \delta_k`.
+  This is the maximum calibration error (MCE) defined in [1].
+
+Here :math:`k` spans all bins,
+:math:`P_k = \dfrac{\sum_{t\in b_k} w_t}{\sum_t w_t}` denotes the (normalized)
+weight of bin :math:`k` and
+:math:`\delta_k = \dfrac{|\sum_{t\in b_k} w_t o_t - \sum_{t\in b_k} w_t f_t|}{\sum_{t\in b_k} w_t}`
+denotes the absolute difference between the average frequency of positive class
+and the average predicted probability of positive class in bin :math:`k`.
+
+The calibration loss is appropriate for binary and categorical outcomes
+that can be structured as true or false, but is inappropriate for ordinal
+variables which can take on three or more values.
+Which label is considered to be the positive label is controlled via the
+parameter pos_label, which defaults to 1.
+
+Here is a small example of usage of this function:::
+
+    >>> import numpy as np
+    >>> from sklearn.metrics import calibration_loss
+    >>> y_true = np.array([0, 0, 0, 1] + [0, 1, 1, 1])
+    >>> y_pred = np.array([0.25, 0.25, 0.25, 0.25] + [0.75, 0.75, 0.75, 0.75])
+    >>> calibration_loss(y_true, y_pred, n_bins=2, norm="l1")
+    0.0
+    >>> calibration_loss(y_true, y_pred, n_bins=2, norm="max")
+    0.0
+    >>> y_true = np.array([0, 0, 0, 0] + [1, 1, 1, 1])
+    >>> calibration_loss(y_true, y_pred, n_bins=2, norm="l1")
+    0.25
+    >>> calibration_loss(y_true, y_pred, n_bins=2, norm="max")
+    0.25
+    >>> from sklearn.metrics import make_scorer
+    >>> from functools import partial
+    >>> max_calibration_loss = partial(calibration_loss, norm = "max")
+    >>> neg_max_calibration_scorer = make_scorer(max_calibration_loss, greater_is_better=False)
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from sklearn.model_selection import GridSearchCV
+    >>> grid = GridSearchCV(RandomForestClassifier(), param_grid={'max_depth': [3, 5, 7]},
+    ...             scoring=neg_max_calibration_scorer)
```

Comment:
Can you please display `grid.best_params_` after this line?

Reply:
Not sure, I understand what do you expect here or what should be highlighted ? I'm happy to add a `make_classification` above and `grid.best_params_` below.

Reply:
Not sure to understand what you mean as there is no `X` to fit the model, and thus no `grid.best_params_` here.
Make up a random `X` to do the `fit` call and then display ?

Reply:
ah indeed you do not fit the model indeed. That's fine as it is, no need to make this code snippet more complex than it is.